CodeEdit/AppDelegate.swift

```diff
@@ -87,27 +87,27 @@ class AppDelegate: NSObject, NSApplicationDelegate, ObservableObject {
 
     // MARK: - Open windows
 
-    @IBAction func openPreferences(_ sender: Any) {
-        if let window = NSApp.windows.filter({ window in
-            return (window.contentView as? NSHostingView<SettingsView>) != nil
-        }).first {
-            window.makeKeyAndOrderFront(self)
-            return
-        }
-
-        let window = NSWindow(
-            contentRect: NSRect(x: 0, y: 0, width: 500, height: 400),
-            styleMask: [.titled, .closable],
-            backing: .buffered, defer: false)
-        window.center()
-        window.toolbar = NSToolbar()
-        window.title = "Settings"
-        window.toolbarStyle = .unifiedCompact
-        _ = NSWindowController(window: window)
-        let contentView = SettingsView()
-        window.contentView = NSHostingView(rootView: contentView)
-        window.makeKeyAndOrderFront(sender)
-    }
+//    @IBAction func openPreferences(_ sender: Any) {
+//        if let window = NSApp.windows.filter({ window in
+//            return (window.contentView as? NSHostingView<SettingsView>) != nil
+//        }).first {
+//            window.makeKeyAndOrderFront(self)
+//            return
+//        }
+//
+//        let window = NSWindow(
+//            contentRect: NSRect(x: 0, y: 0, width: 500, height: 400),
+//            styleMask: [.titled, .closable],
+//            backing: .buffered, defer: false)
+//        window.center()
+//        window.toolbar = NSToolbar()
+//        window.title = "Settings"
+//        window.toolbarStyle = .unifiedCompact
+//        _ = NSWindowController(window: window)
+//        let contentView = SettingsView()
+//        window.contentView = NSHostingView(rootView: contentView)
+//        window.makeKeyAndOrderFront(sender)
+//    }
```

Comment:
Why is this commented out?

Reply:
We don't need to change to SwiftUI lifecycle.

See [this](https://stackoverflow.com/a/60468060/4769852) (approach 1)

Reply:
This is not change to swiftui lifecycel. Because I don't know how to describe them. It like combine the original one and the swiftui one. In this way you can use both swiftui app methods and appkit app AppDelegate. Commented out this is because I use swiftui Setting View. So we don't need this. We are a swiftui view. If all the things we use appkit, it will a little bit strange. We should use the things that SwiftUI have instead of using the appkit one. Also menu bar problem is very simple to be fixed.